### DISCUSSION TITLE
Macro execution

### DISCUSSION THREAD

**Discussion by chross** [tobya/DocTo#190] [Q&A] [answered]
Hi!
I'd like to know how DocTo handles macros and scripting inside Office documents.
Do you know if they will be executed or simply skipped?

**Comment by tobya** [comment, marked as answer]
I dont know perhaps you could try and see what happens.
Let me know if you find out and I will update the docs here.

**Reply by chross** [reply]
Hi! I've tested Word Macros and unfortunately, they are executed.
What I did was:
1) Generate a macro inside a docx document
2) 
```
Sub AutoOpen()
    MsgBox ("AutoOpen")
End Sub
```
3) Save document
4) rename extension from .docx to .doc

When using docto, the macro will be executed.
Can we completely prevent macro execution via docto? I believe this to be a security issue, especially when using docto in an automated unsupervised system.

**Reply by tobya** [reply]
Thanks for confirming.

If you know the VBA code to disable the AutoOpen that would be really helpful.  In the meantime I can put a warning letting people know that any AutoOpen macros present will be executed.

I think most people are using it currently for trusted documents.
